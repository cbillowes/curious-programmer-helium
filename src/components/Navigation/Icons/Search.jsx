import React, { Component } from "react"
import "./Search.scss"

class Search extends Component {
  render() {
    return (
      <div
        className={`icon search ${this.props.active ? "active" : ""}`}
        onClick={this.props.toggle.bind(this)}
      >
        <i />
      </div>
    )
  }
}

export default Search