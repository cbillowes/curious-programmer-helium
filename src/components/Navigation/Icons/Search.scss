.navigation .icons .icon.search i {
    background-image: url(data:image/svg+xml;utf8;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iaXNvLTg4NTktMSI/Pgo8IS0tIEdlbmVyYXRvcjogQWRvYmUgSWxsdXN0cmF0b3IgMTguMS4xLCBTVkcgRXhwb3J0IFBsdWctSW4gLiBTVkcgVmVyc2lvbjogNi4wMCBCdWlsZCAwKSAgLS0+CjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgdmVyc2lvbj0iMS4xIiBpZD0iQ2FwYV8xIiB4PSIwcHgiIHk9IjBweCIgdmlld0JveD0iMCAwIDI1MC4zMTMgMjUwLjMxMyIgc3R5bGU9ImVuYWJsZS1iYWNrZ3JvdW5kOm5ldyAwIDAgMjUwLjMxMyAyNTAuMzEzOyIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSIgd2lkdGg9IjUxMnB4IiBoZWlnaHQ9IjUxMnB4Ij4KPGcgaWQ9IlNlYXJjaCI+Cgk8cGF0aCBzdHlsZT0iZmlsbC1ydWxlOmV2ZW5vZGQ7Y2xpcC1ydWxlOmV2ZW5vZGQ7IiBkPSJNMjQ0LjE4NiwyMTQuNjA0bC01NC4zNzktNTQuMzc4Yy0wLjI4OS0wLjI4OS0wLjYyOC0wLjQ5MS0wLjkzLTAuNzYgICBjMTAuNy0xNi4yMzEsMTYuOTQ1LTM1LjY2LDE2Ljk0NS01Ni41NTRDMjA1LjgyMiw0Ni4wNzUsMTU5Ljc0NywwLDEwMi45MTEsMFMwLDQ2LjA3NSwwLDEwMi45MTEgICBjMCw1Ni44MzUsNDYuMDc0LDEwMi45MTEsMTAyLjkxLDEwMi45MTFjMjAuODk1LDAsNDAuMzIzLTYuMjQ1LDU2LjU1NC0xNi45NDVjMC4yNjksMC4zMDEsMC40NywwLjY0LDAuNzU5LDAuOTI5bDU0LjM4LDU0LjM4ICAgYzguMTY5LDguMTY4LDIxLjQxMyw4LjE2OCwyOS41ODMsMEMyNTIuMzU0LDIzNi4wMTcsMjUyLjM1NCwyMjIuNzczLDI0NC4xODYsMjE0LjYwNHogTTEwMi45MTEsMTcwLjE0NiAgIGMtMzcuMTM0LDAtNjcuMjM2LTMwLjEwMi02Ny4yMzYtNjcuMjM1YzAtMzcuMTM0LDMwLjEwMy02Ny4yMzYsNjcuMjM2LTY3LjIzNmMzNy4xMzIsMCw2Ny4yMzUsMzAuMTAzLDY3LjIzNSw2Ny4yMzYgICBDMTcwLjE0NiwxNDAuMDQ0LDE0MC4wNDMsMTcwLjE0NiwxMDIuOTExLDE3MC4xNDZ6IiBmaWxsPSIjRkZGRkZGIi8+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPGc+CjwvZz4KPC9zdmc+Cg==);
}
