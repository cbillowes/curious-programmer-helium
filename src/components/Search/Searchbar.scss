@import "../../layout/variables.scss";

.searchbar {
    display: none;
    background-color: $background-darker;
    position: fixed;
    left: 0;
    bottom: 0;
    right: 0;
    top: 4em;
    z-index: 1;
    padding: 1em;

    &.active {
        display: block;
    }

    .search-query {
        display: block;
        width: 96%;
        padding: 0.5em;
        margin: 1.5em 2% 0.5em 2%;
        font-family: $font-family-sans-serif;
        background-color: $background-lighter;
        border: solid 1px $background-lighter;
        border-radius: $border-radius;
        color: $foreground;
        font-size: 1.25em;
    }

    .error {
        .message {
            color: $accent;
            font-weight: $font-family-sans-serif-heavy;
        }

        .contact-me {
            a {
                display: inline-block;
            }
        }
    }

    .results {
        height: 100vh;
        overflow-y: auto;
        font-size: $font-size * 1.15;

        ul {
            padding-left: 2%;

            li {
                list-style: none;
                padding-bottom: 0.5em;

                a {
                    display: block;
                }
            }
        }
    }
}
