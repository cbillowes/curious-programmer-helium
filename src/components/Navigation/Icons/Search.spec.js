import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Search from "./Search"

describe("a search icon component", () => {
  let element = <Search toggle={() => console.log("toggle")} active="true" />
  let component
  
  beforeAll(() => {
    component = mount(element)    
  })

  it("should render", () => {
    const tree = renderer.create(element).toJSON()
    expect(tree).toMatchSnapshot()
  })
})