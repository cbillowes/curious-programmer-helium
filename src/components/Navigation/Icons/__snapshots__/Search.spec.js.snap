// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a search icon component should render 1`] = `
<div
  className="icon search active"
  onClick={[Function]}
>
  <i />
</div>
`;
